packages/promotion/integration-tests/__tests__/services/promotion-module/compute-actions.spec.ts

```diff
@@ -2315,4 +2311,366 @@ describe("Promotion Service: computeActions", () => {
       ])
     })
   })
+
+  describe("when promotion of type buyget", () => {
+    it("should compute adjustment when target and buy rules match", async () => {
+      const context = {
+        customer: {
+          customer_group: {
+            id: "VIP",
+          },
+        },
+        items: [
+          {
+            id: "item_cotton_tshirt",
+            quantity: 2,
+            unit_price: 500,
+            product_category: {
+              id: "catg_tshirt",
+            },
+            product: {
+              id: "prod_tshirt_1",
+            },
+          },
+          {
+            id: "item_cotton_tshirt2",
+            quantity: 2,
+            unit_price: 1000,
+            product_category: {
+              id: "catg_tshirt",
+            },
+            product: {
+              id: "prod_tshirt_2",
+            },
+          },
+          {
+            id: "item_cotton_sweater",
+            quantity: 2,
+            unit_price: 1000,
+            product_category: {
+              id: "catg_sweater",
+            },
+            product: {
+              id: "prod_sweater_1",
+            },
+          },
+        ],
+      }
+
+      const [createdPromotion] = await service.create([
+        {
+          code: "PROMOTION_TEST",
+          type: PromotionType.BUYGET,
+          rules: [
+            {
+              attribute: "customer.customer_group.id",
+              operator: "in",
+              values: ["VIP", "top100"],
+            },
+          ],
+          application_method: {
+            type: "fixed",
+            target_type: "items",
+            allocation: "each",
+            max_quantity: 1,
+            apply_to_quantity: 1,
+            buy_rules_min_quantity: 1,
```

Comment:
thought: would want this to be part of the buy_rules instead. e.g., `{ attribute: "quantity", operator: "gt", values: 1 }`. All buy rules would then be evaluated with an `AND` operator on for each item. Wdyt of this idea?

Reply:
I think this might be quite limiting when we want to expand cases. It would only allow one BuyRule for item targeting and will also result in some relationship validations as min_qty is a required field. 

I think there are 2 cases here:

1. each rule needs to conform to a grouped quantity
ex: buy a cotton shirt and a silk shirt and get a sweater for free
or
ex: buy 2 cotton shirt and get a sweater for free

In this case, we need to use an OR operator.

```
buy_rules_type: "across",
buy_rules_min_quantity: 2,
buy_rules: [{
  attribute: "product_category.id",
  operator: "eq",
  values: ["catg_cotton"],
}, {
  attribute: "product_category.id",
  operator: "eq",
  values: ["catg_silk"],
}],
```

2. each rule conforms to its own min_quantity
ex: buy 2 cotton tshirts and 1 silk shirt and get a sweater free.


In this case, we need to use an AND operator.

```
buy_rules_type: "each",
buy_rules_min_quantity: null,
buy_rules: [{
  attribute: "product_category.id",
  operator: "eq",
  values: ["catg_cotton"],
  min_quantity: 2 
}, {
  attribute: "product_category.id",
  operator: "eq",
  values: ["catg_silk"],
  min_quantity: 1 
}],
```

wdyt?

Reply:
That makes sense! Kinda hard to wrap your head around, but it makes sense - would be amazing if this could self-document somehow.

Reply:
Currently only case 1 is implemented. When implementing case 2, I can rework the code to make it more easily readable with some inline comments.